tests/integration/shell/auth.py

```diff
@@ -7,18 +7,37 @@
 # Import python libs
 import os
 import pwd
+import grp
 import random
 
 # Import Salt Testing libs
+from salttesting import skipIf
 from salttesting.helpers import (
     ensure_in_syspath,
     destructiveTest)
 ensure_in_syspath('../../')
 
 # Import salt libs
+from salt.utils.pycrypto import gen_hash
 import integration
 
-from salttesting import skipIf
+
+def gen_password():
+    '''
+    generate a password and hash it
+    '''
+    alphabet = ('abcdefghijklmnopqrstuvwxyz'
+                '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ')
+    password = ''
+    # generate password
+    for _ in range(20):
+        next_index = random.randrange(len(alphabet))
+        password += alphabet[next_index]
```

Comment:
Why `seed` is omitted? And also this can be written a bit better:
```python
random.seed = os.urandom(0x400)
password = ''.join((random.choice(alphabet) for i in range(20)))
```